src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -515,110 +452,169 @@ public void updateSubscribers()
+        {
+            subscribers
+                    .stream()
+                    .filter(player -> permissions.isDirty() || !oldSubscribers.contains(player)).forEach(player -> {
+                Permissions.Rank rank = getPermissions().getRank(player);
+                MineColonies.getNetwork().sendTo(new PermissionsMessage.View(this, rank), player);
+            });
+        }
+    }
+
+    private void sendColonyViewPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
+    {
+        if (isDirty || hasNewSubscribers)
+        {
+            for (EntityPlayerMP player : subscribers)
             {
-                for (CitizenData citizen : citizens.values())
+                boolean isNewSubscriber = !oldSubscribers.contains(player);
+                if (isDirty || isNewSubscriber)
                 {
-                    if (citizen.isDirty() || hasNewSubscribers)
-                    {
-                        ColonyViewCitizenViewMessage msg = new ColonyViewCitizenViewMessage(this, citizen);
-
-                        subscribers.stream()
-                                   .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
-                                   .forEach(player -> MineColonies.getNetwork().sendTo(msg, player));
-                    }
+                    MineColonies.getNetwork().sendTo(new ColonyViewMessage(this, isNewSubscriber), player);
                 }
             }
+        }
+    }
 
-            //  Buildings
-            if (isBuildingsDirty || hasNewSubscribers)
+    private static boolean hasNewSubscribers(Set<EntityPlayerMP> oldSubscribers, Set<EntityPlayerMP> subscribers)
+    {
+        for (EntityPlayerMP player : subscribers)
+        {
+            if (!oldSubscribers.contains(player))
             {
-                for (Building building : buildings.values())
-                {
-                    if (building.isDirty() || hasNewSubscribers)
-                    {
-                        ColonyViewBuildingViewMessage msg = new ColonyViewBuildingViewMessage(building);
+                return true;
+            }
+        }
+        return false;
+    }
 
-                        subscribers.stream()
-                                   .filter(player -> building.isDirty() || !oldSubscribers.contains(player))
-                                   .forEach(player -> MineColonies.getNetwork().sendTo(msg, player));
-                    }
+    private static boolean isCitizenMissingFromWorld(CitizenData citizen)
+    {
+        EntityCitizen entity = citizen.getCitizenEntity();
+
+        return entity != null && entity.worldObj.getEntityByID(entity.getEntityId()) != entity;
+    }
+
+    private boolean areAllColonyChunksLoaded(TickEvent.WorldTickEvent event)
+    {
+        int distanceFromCenter = Configurations.workingRangeTownHall + 48 /* 3 chunks */ + 15 /* round up a chunk */;
+        for (int x = -distanceFromCenter; x <= distanceFromCenter; x += 16)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)